tests/modcc/test_kinetic_rewriter.cpp

```diff
@@ -0,0 +1,507 @@
+#include "test.hpp"
+
+#include <cmath>
+#include <stdexcept>
+#include <string>
+
+#include "expression.hpp"
+#include "kinrewriter.hpp"
+#include "parser.hpp"
+
+// Simple algebraic term expansion/collection routines.
+
+namespace alg {
```

Comment:
These algorithms look quite useful. What do you think of moving them to a separate header file inside test?